### PR TITLE
args: Honor config-dir & quiet wherever they are

### LINKED ISSUE #3354: server: --config-dir and --quiet should be honored when placed after 'server' argument
## Expected Behavior
./minio server --config-dir /home/me/config ./photos should load config from /home/me/config

## Current Behavior
Still using ~/.minio

## Your Environment
All

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
setGlobalsFromContext() is added to set global variables after parsing
command line arguments. Thus, global flags will be honored wherever
they are placed in the command.

## Motivation and Context
Fixes #3354 

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -148,72 +147,86 @@ func checkMainSyntax(c *cli.Context) {
...
-		// Sets new config directory.
-		setGlobalConfigPath(configDir)
-
-		// Valid input arguments to main.
-		checkMainSyntax(c)
-
-		// Migrate any old version of config / state files to newer format.
-		migrate()
+// Parse command arguments and set global variables accordingly
+func setGlobalsFromContext(c *cli.Context) {
```
Comment (harshavardhana) [inline cmd/main.go]: Move this function to `globals.go`

File: cmd/server-main.go [hunk 2]
```diff
@@ -363,8 +363,13 @@ func serverMain(c *cli.Context) {
...
 	}
 
-	// Set global quiet flag.
-	globalQuiet = c.Bool("quiet") || c.GlobalBool("quiet")
+	// Set global variables after parsing passed arguments
+	setGlobalsFromContext(c)
+
+	checkUpdate()
+
+	// Generic initialization
```
Comment (harshavardhana) [inline cmd/server-main.go]: Comment should talk about what it does.. "Generic initialization" is perhaps too "Generic"

File: cmd/update-main.go [hunk 3]
```diff
@@ -265,8 +265,11 @@ func getReleaseUpdate(updateURL string, duration time.Duration) (updateMsg updat
 
 // main entry point for update command.
 func mainUpdate(ctx *cli.Context) {
-	// Set global quiet flag.
-	if ctx.Bool("quiet") || ctx.GlobalBool("quiet") {
+
+	setGlobalsFromContext(ctx)
```
Comment (harshavardhana) [inline cmd/update-main.go:270]: Comment is needed..

File: cmd/version-main.go [hunk 4]
```diff
@@ -43,11 +43,16 @@ func mainVersion(ctx *cli.Context) {
 	if len(ctx.Args()) != 0 {
 		cli.ShowCommandHelpAndExit(ctx, "version", 1)
 	}
-	// Set global quiet flag.
-	if ctx.Bool("quiet") || ctx.GlobalBool("quiet") {
+
+	setGlobalsFromContext(ctx)
+	minioInit()
```
Comment (harshavardhana) [inline cmd/version-main.go:51]: Comment is needed..

File: cmd/version-main.go [hunk 5]
```diff
@@ -43,11 +43,16 @@ func mainVersion(ctx *cli.Context) {
...
+
+	if globalQuiet {
 		return
 	}
 
+	checkUpdate()
```
Comment (harshavardhana) [inline cmd/version-main.go]: No need to checkUpdate() for looking for `version`

**Comment by harshavardhana** [comment]
Any reviews here @krisis @krishnasrinivas?

**Review by krisis — APPROVED** [review]
Leaving aside @harshavardhana comment regarding calling ``checkUpdate()`` for minio version subcommand, the PR looks good to me.

**Comment by vadmeste** [comment]
> Leaving aside @harshavardhana comment regarding calling checkUpdate() for minio version subcommand, the PR looks good to me.

Done, thanks.

**Pull request merged**